R/add_styles.R

```diff
@@ -0,0 +1,77 @@
+#' List of all available styles
+#'
+#' A list containing all available styling options for hierplane
+#'
+#' Source: \url{https://github.com/allenai/hierplane/}
+"styles"
+
+#' Add styles to hierplane settings
+#'
+#' Append styling options to output of `hierplane_settings()`.
+#' @param settings A list of settings from `hierplane_settings()`
+#' @param node_type_to_style TODO
+#' @param link_to_positions TODO
+#' @param link_name_to_label TODO
+#'
+#' @export
+
+add_styles <- function(settings,
+                       node_type_to_style = NULL,
+                       link_to_positions = NULL,
+                       link_name_to_label = NULL) {
+
+  settings$node_type_to_style <- format_styles(node_type_to_style, "node_type_to_style")
+  settings$link_to_positions <- format_styles(link_to_positions, "link_to_positions")
+  settings$link_name_to_label <- format_styles(link_name_to_label, "link_name_to_label")
+
+  settings
+}
+
+format_styles <- function(x, name) {
+
+  if (is.null(x)) return(NULL)
+  if (!"list" %in% class(x)) stop(paste(name, "must be a list."), call. = F)
+
+  if (name %in% "node_type_to_style") {
+    return(lapply(x, function(x) if (!is.list(x)) as.list(x) else x))
+  } else {
+    return(x)
+  }
+
+}
+
+check_style <- function(x, settings, setting_type, setting_target) {
+
+  selected <- settings[[setting_type]]
+  bad_vals <- setdiff(names(selected), x[[settings[[setting_target]]]])
+  bad_settings <- setdiff(unlist(selected), style[[setting_type]])
```

Comment:
Should this be `x[[setting_type]]` instead of `style[[setting_type]]`?

Reply:
This is trying to check the values selected in the settings against the valid options defined in `styles`. I think it should be `styles[[setting_type]]`. Also while typing this I realized I made a typo and it should be `styles` instead of `style`. whoops

Reply:
Got it, this once you change to `styles`, RCMD should pass 😸

Reply:
🤦

Reply:
Sorry, I was wrong, pushing the fix.

Reply:
No it's my bad. I realized after I checked the rcmd that I never hit ctrl+d 🤦 🤦 🤦 

By the time I tried to push the changes you already fixed it. hehehhe thanks!